### PR TITLE
staticanalysis/bot: Do not check clang-tidy exit code anymore

### PR DESCRIPTION
**Pull request by La0**
From irc #release-mgmt-analysis

> Andi> | starting with clang7 all of the errors that are assess by clang-tidy will make the process now to return a non-0 return code

So the bot must **not** check the exit code anymore, and simply rely on parsing the output

### CODE REVIEW

File: src/staticanalysis/bot/static_analysis_bot/clang/tidy.py [hunk 1]
```diff
@@ -105,13 +105,14 @@ def run(self, revision):
         ] + list(revision.files)
         logger.info('Running static-analysis', cmd=' '.join(cmd))
 
-        # Run command
+        # Run command, without checking its exit code as clang-tidy 7+
+        # exits with an error code when finding errors (which we want to report !)
         try:
```
Comment (marco-c) [inline src/staticanalysis/bot/static_analysis_bot/clang/tidy.py]: Is the exception handling still necessary? Will this change hide other possible errors?

File: src/staticanalysis/bot/static_analysis_bot/clang/tidy.py [hunk 2]
```diff
@@ -105,13 +105,14 @@ def run(self, revision):
...
-            clang_output = subprocess.check_output(cmd, cwd=settings.repo_dir)
+            clang = subprocess.run(cmd, cwd=settings.repo_dir, check=False, stdout=subprocess.PIPE)
```
Comment (marco-c) [inline src/staticanalysis/bot/static_analysis_bot/clang/tidy.py]: Nit: `check` is false by default

Comment (marco-c) [inline src/staticanalysis/bot/static_analysis_bot/clang/tidy.py]: Nit: `capture_output` could be used instead of stdout=subprocess.PIPE

**Comment by La0** [comment]
@marco-c A unit test is already triggerring the clang-tidy run method (hence the update in this PR), but the real call to clang-tidy is mocked.

I agree on point 2, i really don't like this solution

**Comment by jankeromnes** [comment]
> 1. We should add a test for this;
> 2. I'm afraid we might start missing some issues where "mach static analysis" fails for other reasons

Agreed, @marco-c is right

**Comment by marco-c** [comment]
> @marco-c A unit test is already triggerring the clang-tidy run method (hence the update in this PR), but the real call to clang-tidy is mocked.

Do we also have a test where clang-tidy returns a non-0 error code?

**Comment by abpostelnicu** [comment]
There is another solution, to directly patch clang-tidy when we build the artifact, I can do this if you guys thin it's a better way. This is where we return 1.
https://github.com/llvm-mirror/clang-tools-extra/commit/923585d864cc22bbab6569f461b61ddf31b2d9c4#diff-3f59c90026cc7a6567edcfc58d69530eR493

**Comment by abpostelnicu** [comment]
A followup bug for this issue in clang-tidy - https://treeherder.mozilla.org/#/jobs?repo=try&revision=e1213a3d6b9c4c0c481301099fff81cc7a766d0b

**Pull request merged**